lib/src/view/game/analysis_screen.dart

```diff
@@ -0,0 +1,944 @@
+
+    // add the node[0]
+    final newPath = initialPath + nodes[0].id;
+    final currentMove = newPath == currentPath;
+    widgets.add(
+      InlineMove(
+        ctrlProvider,
+        path: newPath,
+        move: nodes[0].sanMove,
+        ply: nodes[0].ply,
+        isCurrentMove: currentMove,
+        key: currentMove ? currentMoveKey : null,
+        isSideline: !inMainline,
+        startSideline: startSideline,
+        endSideline: !inMainline && nodes[0].children.isEmpty,
+      ),
+    );
+
+    // add the sidelines if present
+    for (var i = 1; i < nodes.length; i++) {
+      // start new sideline from mainline on a new line
+      if (inMainline) {
+        widgets.add(
+          SizedBox(
+            width: double.infinity,
+            child: Wrap(
+              spacing: 1.0,
+              children: _buildTreeWidget(
+                ctrlProvider,
+                nodes: IList([nodes[i]]),
+                inMainline: false,
+                startSideline: true,
+                initialPath: initialPath,
+                currentPath: currentPath,
+              ),
+            ),
+          ),
+        );
+      } else {
+        widgets.addAll(
+          _buildTreeWidget(
+            ctrlProvider,
+            nodes: IList([nodes[i]]),
+            inMainline: false,
+            startSideline: true,
+            initialPath: initialPath,
+            currentPath: currentPath,
+          ),
+        );
+      }
+    }
+
+    // add the children of the first child
+    widgets.addAll(
+      _buildTreeWidget(
+        ctrlProvider,
+        nodes: nodes[0].children,
+        inMainline: inMainline,
+        startSideline: false,
+        initialPath: newPath,
+        currentPath: currentPath,
+      ),
+    );
+
+    return widgets;
+  }
+}
+
+class InlineMove extends ConsumerWidget {
+  const InlineMove(
+    this.ctrlProvider, {
+    required this.path,
+    required this.move,
+    required this.ply,
+    required this.isCurrentMove,
+    required this.isSideline,
+    super.key,
+    this.startSideline = false,
+    this.endSideline = false,
+  });
+
+  final AnalysisCtrlProvider ctrlProvider;
+  final UciPath path;
+  final SanMove move;
+  final int ply;
+  final bool isCurrentMove;
+  final bool isSideline;
+  final bool startSideline;
+  final bool endSideline;
+  static const borderRadius = BorderRadius.all(Radius.circular(8.0));
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final baseStyle = isSideline ? sideLineStyle : baseTextStyle;
+
+    final index = ply.isOdd
+        ? Text(
+            '${(ply / 2).ceil()}. ',
+            style: baseStyle,
+          )
+        : (startSideline
+            ? Text(
+                '${(ply / 2).ceil()}... ',
+                style: baseStyle,
+              )
+            : null);
+
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        if (startSideline)
+          const Opacity(
```

Comment:
`Opacity` widget in flutter is expensive: https://api.flutter.dev/flutter/widgets/Opacity-class.html

Here I think we can use instead the opacity of the font color.

The comment applies for all `Opacity` found here.